eyereturn/eyereturn-htb.js

```diff
@@ -0,0 +1,328 @@
+'use strict';
+
+var Browser = require('browser.js');
+var Classify = require('classify.js');
+var Constants = require('constants.js');
+var Partner = require('partner.js');
+var Size = require('size.js');
+var SpaceCamp = require('space-camp.js');
+var System = require('system.js');
+var Network = require('network.js');
+
+var RenderService;
+
+//? if (DEBUG) {
+var ConfigValidators = require('config-validators.js');
+var PartnerSpecificValidator = require('eyereturn-htb-validator.js');
+var Scribe = require('scribe.js');
+var Whoopsie = require('whoopsie.js');
+//? }
+
+/**
+ * Partner module template
+ *
+ * @class
+ */
+function EyereturnHtb(configs) {
+    /**
+     * Reference to the partner base class.
+     *
+     * @private {object}
+     */
+    var __baseClass;
+
+    /**
+     * Profile for this partner.
+     *
+     * @private {object}
+     */
+    var __profile;
+
+    /**
+     * Generates the request URL and query data to the endpoint for the xSlots
+     * in the given returnParcels.
+     *
+     * @param  {object[]} returnParcels
+     *
+     * @return {object}
+     */
+    function __generateRequestObj(returnParcels) {
+        var callbackId = System.generateUniqueId();
+
+        // MRA only get one parcel
+        var returnParcel = returnParcels[0];
+        var xSlot = returnParcel.xSlotRef;
+        var payload = {};
+        // eslint-disable-next-line camelcase
+        payload.ad_slot = [];
+
+        // Get the dimensions
+        for (var i = 0; i < xSlot.sizes.length; i++) {
+            var size = xSlot.sizes[i];
+            if (size.length === 2) {
+                payload.ad_slot.push({
+                    width: size[0],
+                    height: size[1]
+                });
+            }
+        }
+
+        // eslint-disable-next-line camelcase
+        payload.request_id = returnParcel.requestId;
+
+        payload.site = {
+            url: Browser.getPageUrl(),
+            referrer: Browser.getReferrer(),
+            ppid: xSlot.ppid
+        };
+
+        var baseUrl = 'https://prometheus-ix.eyereturn.com/prometheus/bid';
+
+        return {
+            url: baseUrl,
+            data: payload,
+            callbackId: callbackId,
+            networkParamOverrides: {
+                method: 'POST',
+                contentType: 'application/json'
+            }
+        };
+    }
+
+    function adResponseCallback(adResponse) {
```

Comment:
If `__profile.callbackType` is set to `CALLBACK_NAME` or `NONE`, the wrapper will handle the callback for you and you can remove this function. ( Reference: https://github.com/ermi-adapter/ht-wrapper-adapters --> Step 3: Response Callback)